--- Makefile	2009-12-17 17:20:27.000000000 +0100
+++ Makefile.oden	2010-04-27 13:56:11.798328493 +0200
@@ -20,22 +20,26 @@
 # along with MMC.  If not, see <http://www.gnu.org/licenses/>.
 
 CONFFILE = /etc/openldap/mmc-check-password.conf
+CFLAGS = -g -O2 -Wall -fpic
+DESTDIR = 
 
 all: mmc-check-password
 
 mmc-check-password.o:
-	$(CC) -g -O2 -Wall -fpic -c -I/usr/include/openldap/include -I/usr/include/openldap/slapd -DCONFIG_FILE=\"$(CONFFILE)\" mmc-check-password.c
+	$(CC) $(CFLAGS) -c -I/usr/include/openldap/include -I/usr/include/openldap/slapd -DCONFIG_FILE=\"$(CONFFILE)\" mmc-check-password.c
 
 mmc-check-password: clean mmc-check-password.o
-	$(CC) -shared -o mmc-check-password.so mmc-check-password.o
+	$(CC) $(CFLAGS) -shared -Wl,-soname=mmc-check-password.so -o \
+	mmc-check-password.so mmc-check-password.o
 
 install: mmc-check-password
 	# FIXME: don't know how to make it better ...
-	-[ -d /usr/lib/openldap ] && \
-	$(INSTALL) mmc-check-password.so -m 755 -o root -g root /usr/lib/openldap/
-	-[ -d /usr/lib64/openldap ] && \
-	$(INSTALL) mmc-check-password.so -m 755 -o root -g root /usr/lib64/openldap/
-	$(INSTALL) mmc-check-password.conf -m 644 -o root -g root $(CONFFILE)
+	-[ -d $(DESTDIR)/usr/lib/openldap ] && \
+	$(INSTALL) mmc-check-password.so -m 755 $(DESTDIR)/usr/lib/openldap/
+	-[ -d $(DESTDIR)/usr/lib64/openldap ] && \
+	$(INSTALL) mmc-check-password.so -m 755 $(DESTDIR)/usr/lib64/openldap/
+	$(INSTALL) -d -m 755 $(DESTDIR)/etc/openldap
+	$(INSTALL) mmc-check-password.conf -m 644 $(DESTDIR)$(CONFFILE)
 
 clean:
 	$(RM) mmc-check-password.o mmc-check-password.so *~
